build-push-docker-image/action.yml

```
@@ -0,0 +1,59 @@
+name: Build and Push docker image
+description: Build and Push docker image
+
+inputs:
+  poetry_version:
+    description: "Poetry version to install"
+    required: true
+  python_version:
+    description: "Python version to use"
+    required: true
+  image_tag:
+    description: "Docker image tag to use. Defaults to repository name + github sha"
+    required: false
+    default: "${{ github.event.repository.name }}-${{ github.sha }}"
+  is_pushable:
+    description: "Depicts if the image should be pushed"
+    required: false
+    default: 'true'
+  ecr_repository:
+    description: "ECR Repository to push image in. Defaults to repository name"
+    required: false
+    default: ${{ github.event.repository.name }}
+
+runs:
+  using: composite
+  steps:
+    - name: Build Image
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: |
+        ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+        ssh-add - <<< "$SSH_KEY"
+        docker build \
+            --build-arg PYTHON_VERSION_CONSTRAINT="${{ inputs.python_version }}" \
+            --build-arg POETRY_VERSION="${{ inputs.poetry_version }}" \
+            --ssh default=$SSH_AUTH_SOCK -t "${{ inputs.image_tag }}" .
+      shell: bash
+
+    - if: inputs.is_pushable == 'true' && github.ref == 'refs/heads/master'
```

We can use `github.ref_name` which was introduced some time ago.
```suggestion
    - if: inputs.push_to_ecr == 'true' && github.ref_name == 'master'
```